### PR TITLE
User auth

### PR DESCRIPTION
**Pull request by richrliu**

### CODE REVIEW

File: inventory/app.py [hunk 1]
```diff
@@ -83,6 +84,66 @@ def seed_test_data():
...
     return jsonify("Successfully seeded data")
 
+@app.route('/user/requestAccess', methods=['POST'])
+def userRequestAccess():
+    content = flask_request.get_json()
+    net_id = content.get('net_id')
+    username = content.get('username')
+    password = content.get('password')
+    role_id = content.get('role_id')
+    if username is None or password is None or net_id is None or role_id is None:
```
Comment (danrili) [inline inventory/app.py]: `if not all([username, password, net_id, role_id])`:

File: inventory/app.py [hunk 2]
```diff
@@ -83,6 +84,66 @@ def seed_test_data():
...
+            first_name = "",
+            last_name = "",
+            role_id = role_id #TODO Fix
+        );
+        user.hashPassword(password)
+        db.session.add(user)
+        db.session.commit()
+    return jsonify("User successfully requested access")
+
+# TODO: Gate to admin only
```
Comment (danrili) [inline inventory/app.py]: Make a decorator that wraps all authed endpoints. the decorator can take in a property like a list of role ids that have whitelisted access to the endpoint

File: inventory/static/scripts/jsx/createUsersForm.jsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
+var CreateUsersForm = React.createClass({
+  render: () => {
+    return (
+      <div>
+        <p>To add users to the system, enter their NetIDs separated by commas and specify their role</p>
+        <div className="input-group">
+          <input type="text" className="form-control" placeholder="Duke NetID's" aria-describedby="basic-addon2" />
```
Comment (danrili) [inline inventory/static/scripts/jsx/createUsersForm.jsx]: not sure what `aria-describedby="basic-addon2"` is

File: inventory/templates/createUsers.html [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  <head>
+    <link rel="stylesheet" href="{{ url_for('static', filename='bower_components/bootstrap/dist/css/bootstrap.css') }}">
+  </head>
+  <body>
+    <div id="main"><div>
+    <!-- scripts -->
+    <script src="{{ url_for('static', filename='bower_components/react/react.min.js') }}"></script>
+    <script src="{{ url_for('static', filename='bower_components/react/react-dom.min.js') }}"></script>
+    <script src="{{ url_for('static', filename='bower_components/bootstrap/dist/js/bootstrap.min.js') }}"></script>
+    <script src="{{ url_for('static', filename='scripts/js/createUsers.js') }}"></script>
```
Comment (danrili) [inline inventory/templates/createUsers.html]: everything will be bundled into bundle.js import that plz

File: inventory/templates/shibLogin.html [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+<html>
+  <head>
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
+  </head>
+  <body>
+    <div id="main"><div>
+    <!-- scripts -->
+    <script src="{{ url_for('static', filename='bower_components/react/react.min.js') }}"></script>
+    <script src="{{ url_for('static', filename='bower_components/react/react-dom.min.js') }}"></script>
+    <script src="{{ url_for('static', filename='scripts/js/shibLogin.js') }}"></script>
```
Comment (danrili) [inline inventory/templates/shibLogin.html]: same. use bundle.js

File: inventory/static/scripts/jsx/createUsersForm.jsx (hunk 3 above)
Comment (richrliu) [inline inventory/static/scripts/jsx/createUsersForm.jsx]: Some bootstrap thing that we dont have to use

**Pull request closed without merging**